tests/manage/cluster/test_must_gather.py

```diff
@@ -0,0 +1,180 @@
+import os
+import logging
+import time
+
+import pytest
+
+from ocs_ci.framework.testlib import ManageTest, tier1, bugzilla
+from ocs_ci.ocs import openshift_ops, ocp
+from ocs_ci.ocs.utils import collect_ocs_logs
+
+logger = logging.getLogger(__name__)
+
+
+@tier1
+@pytest.mark.polarion_id("OCS-1583")
+@bugzilla('1766646')
+class TestMustGather(ManageTest):
+
+    @pytest.fixture(autouse=True)
+    def init_ocp(self):
+        """
+        init OCP() object
+        """
+
+        self.ocs = openshift_ops.OCP()
+        self.ocp_obj = ocp.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+            if must_gather_pods:
+                for pod_to_del in must_gather_pods:
+                    self.ocp_obj.wait_for_delete(resource_name=pod_to_del)
```

Comment:
We are not setting up the pods, why delete them?

Reply:
it seems that TestManage tear down does not delete this pod in time, so this is a double check

Reply:
@vasukulkarni if I'm not mistaken, a pod is being created for must gather logs collection. It's being destroyed when must gather is done. So here, we wait for that pod to get deleted so leftover checks will be happy

Reply:
Thanks, I see that now.